src/components/vm/vmActions.tsx

```diff
@@ -93,115 +137,69 @@ const onInstall = (vm: VM) => (
-    domainPause({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to pause"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmPause(vm: VM) {
+    await actionWithoutProgress(
+        vm,
+        () => vmDomainMethod<void>(vm, 'Suspend', ''),
+        cockpit.format(_("VM $0 failed to pause"), vm.name),
+    );
+}
 
-const onResume = (vm: VM) => (
-    domainResume({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to resume"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmResume(vm: VM) {
+    await actionWithoutProgress(
+        vm,
+        () => vmDomainMethod<void>(vm, 'Resume', ''),
+        cockpit.format(_("VM $0 failed to resume"), vm.name),
+    );
+}
 
-const onForceoff = (vm: VM) => (
-    domainForceOff({ id: vm.id, connectionName: vm.connectionName })
-            .then(() => !vm.persistent && cockpit.location.go(["vms"]))
-            .catch(ex => {
-                store.dispatch(
-                    updateVm({
-                        connectionName: vm.connectionName,
-                        name: vm.name,
-                        error: {
-                            text: cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
-                            detail: ex.message,
-                        }
-                    })
-                );
-            })
-);
+export async function vmForceOff(vm: VM) {
+    await actionWithoutProgress(
+        vm,
+        async () => {
+            await vmDomainMethod<void>(vm, 'Destroy', 'u', 0);
+            if (!vm.persistent)
+                cockpit.location.go(["vms"]);
+        },
+        cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
+    );
+}
 
-const onSendNMI = (vm: VM) => (
-    domainSendNMI({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function onSendNMI(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'InjectNMI', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to send NMI"), vm.name), ex);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-1714e4ac-20251027-140501-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)